zerver/lib/event_schema.py

```diff
@@ -977,10 +977,20 @@ def check_realm_default_update(
     ]
 )
 
-message_edit_data = DictType(
+allow_message_editing_data = DictType(
     required_keys=[
         ("allow_message_editing", bool),
-        ("message_content_edit_limit_seconds", int),
+    ]
+)
+
+message_content_edit_limit_seconds_data = DictType(
+    required_keys=[
+        ("message_content_edit_limit_seconds", OptionalType(int)),
```

Comment:
Why is this one `Optional`?

Reply:
We need this since the setting can be `None` for case when there is no limit. Previously it worked fine because we did not test `None` case, now also added a test for it.

Reply:
OK great, thanks for confirming the previous state was a bug.